libpcsxcore/psxbios.c

```diff
@@ -788,9 +788,19 @@ void psxBios_bcmp() { // 0x29
 
 void psxBios_memcpy() { // 0x2a
 	char *p1 = (char *)Ra0, *p2 = (char *)Ra1;
-	while ((s32)a2-- > 0) *p1++ = *p2++;
-
-	v0 = a0; pc0 = ra;
+	s32 n=0;
+	v0 = a0;
+	if (a0 == 0 || a2 > 0x7FFFFFFF)
+	{
+		pc0 = ra;
+		return;
+	}
+	while ((s32)a2-- > 0) {
+		n++;
```

Comment:
What is this 'n' for?

Reply:
Hmm i took it directly from pcsx4all (dmitrysmagin did some work on the HLE after all) and i'm not sure why he did this. In fact, nothing in the doc says that this should be done this way. (it's mostly straightforward). So i removed it since it's not being used anyway.